lib/stdlib/test/io_SUITE.erl

```diff
@@ -2947,3 +2948,108 @@ otp_15847(_Config) ->
 otp_15875(_Config) ->
     S = io_lib:format("~tp", [[{0, [<<"00">>]}]], [{chars_limit, 18}]),
     "[{0,[<<48,...>>]}]" = lists:flatten(S).
+
+error_info(_Config) ->
+
+    FullDev =
+        fun() ->
+                {ok, Dev} = file:open("/dev/full",[read, write]),
```

Comment:
This might only work on Linux. It certainly does not work on macOS or Windows.

I think that it would be better to create a fake IO device (as an Erlang process that implements the IO protocol). We might have code for a fake IO devices in some other test suites.

Reply:
> This might only work on Linux. It certainly does not work on macOS or Windows.

Indeed, I need to add a check for that.

> I think that it would be better to create a fake IO device (as an Erlang process that implements the IO protocol). We might have code for a fake IO devices in some other test suites.

Yes and no. If we change how file_io_server returns errors, we should have a test for that. But your idea about creating a dummy io server could help test some other faults that I was unable to test with the built-in versions.